lib/dslash_ndeg_twisted_clover_preconditioned.cu

```diff
@@ -15,15 +15,12 @@
 namespace quda
 {
   // trait to ensure we don't instantiate asymmetric & xpay
-  // defined dslash_ndeg_twisted_mass_preconditioned.cu
-  template <bool symmetric> constexpr bool xpay_();
-  template <> constexpr bool xpay_<true>();
+  template <bool symmetric> constexpr bool xpay_() { return true; }
+  template <> constexpr bool xpay_<true>() { return false; }
```

Comment:
why is this allowed, these functions are already defined in the namespace quda...

Reply:
const and constexpr have internal linkage, these functions don't exist beyond file-scope

Reply:
![image](https://user-images.githubusercontent.com/1325246/111767533-236a0a00-88a7-11eb-8889-aa4cc9c43627.png)